regression/rt_ar_react_1x2v_quadraticV.c

```diff
@@ -0,0 +1,969 @@
+      strcpy(reactions[j].ionization[i].elc_nm, electron_name);
+      reactions[j].ionization[i].charge_state = i;
+      reactions[j].ionization[i].ion_mass = ion_mass;
+      reactions[j].ionization[i].elc_mass = electron_mass;
+      strcpy(reactions[j].ionization[i].ion_nm, ion_nm);
+      strcpy(reactions[j].ionization[i].donor_nm, donor_nm);
+      reactions[j].ionization[i].type_self = j;
+    }
+  }
+  return reactions;
+}
+
+void
+write_data(struct gkyl_tm_trigger *iot, gkyl_gyrokinetic_app *app, double tcurr)
+{
+  if (gkyl_tm_trigger_check_and_bump(iot, tcurr)) {
+    gkyl_gyrokinetic_app_write(app, tcurr, iot->curr-1);
+    gkyl_gyrokinetic_app_calc_mom(app); gkyl_gyrokinetic_app_write_mom(app, tcurr, iot->curr-1);
+  }
+}
+
+int  
+main(int argc, char **argv)
+{
+  struct gkyl_app_args app_args = parse_app_args(argc, argv);
+
+  if (app_args.trace_mem) {
+    gkyl_cu_dev_mem_debug_set(true);
+    gkyl_mem_debug_set(true);
+  }
+
+  struct gk_app_ctx ctx = create_ctx(); // context for init functions
+
+  // Reactions for argon isonuclear sequence
+  struct all_of_1species_reactions *reactions = create_isonuclear_react(ctx.massAr, ctx.massElc, 18, GKYL_ION_AR, "elc", "Ar");
+
+  int NX = APP_ARGS_CHOOSE(app_args.xcells[0], 2);
+  int NV = APP_ARGS_CHOOSE(app_args.vcells[0], 6);
+  int NMU = APP_ARGS_CHOOSE(app_args.vcells[1], 4);
+
+  // electrons
+  struct gkyl_gyrokinetic_species elc = {
+    .name = "elc",
+    .charge = ctx.chargeElc, .mass = ctx.massElc,
+    .lower = { -1.0, 0.0},
+    .upper = { 1.0, 1.0}, 
+    .cells = { NV, NMU },
+    .polarization_density = ctx.n0,
+
+    .mapc2p = {
+      .mapping = mapc2p_vel_elc,
+      .ctx = &ctx,
+    },
+
+    .projection = {
+      .proj_id = GKYL_PROJ_MAXWELLIAN_PRIM, 
+      .ctx_density = &ctx,
+      .density = eval_density,
+      .ctx_upar = &ctx,
+      .upar= eval_upar,
+      .ctx_temp = &ctx,
+      .temp = eval_temp_elc,      
+    },
+
+     .collisions =  {
+      .collision_id = GKYL_LBO_COLLISIONS,
+      .normNu = true,
+      .n_ref = ctx.n0,
+      .T_ref = ctx.Te,      
+      .self_nu = evalNuElc,
+      .ctx = &ctx,
+      .num_cross_collisions = 3,
+      .collide_with = { "ion", "Ar1", "Ar2" },
+      },
+
+    .react = {
+      .num_react = 2,
+      .react_type = {
+        { .react_id = GKYL_REACT_IZ, 
+          .type_self = GKYL_SELF_ELC, 
+          .ion_id = GKYL_ION_AR, 
+          .elc_nm = "elc", 
+          .ion_nm = "Ar2", 
+          .donor_nm = "Ar1", 
+          .charge_state = 1, 
+          .ion_mass = ctx.massAr, 
+          .elc_mass = ctx.massElc, 
+        }, 
+        { .react_id = GKYL_REACT_RECOMB,
+          .type_self = GKYL_SELF_ELC,
+          .ion_id = GKYL_ION_AR,
+          .elc_nm = "elc",
+          .ion_nm = "Ar2",
+          .recvr_nm = "Ar1",
+          .charge_state = 1,
+          .ion_mass = ctx.massAr,
+          .elc_mass = ctx.massElc,
+        },
+      },
+      },
+    
+    /*.bcx = {
```

Comment:
delete commented code, or describe in comments with your initials and date why this commented code should really be kept. Same for other commented code below and elsewhere